src/m365/commands/login.spec.ts

```diff
@@ -296,6 +296,35 @@ describe(commands.LOGIN, () => {
     assert.strictEqual(auth.service.authType, AuthType.Browser, 'Incorrect authType set');
   });
 
+  it(`Don't start login flow when the CLI is signed in`, async () => {
+    sinon.stub(auth, 'ensureAccessToken').callsFake(() => Promise.resolve(''));
+
+    await command.action(logger, { options: { ensure: true } });
+    assert.strictEqual(auth.service.connected, true);
```

Comment:
How are we verifying that the login flow hasn't started? Aren't we only checking that the user is signed in in the end, which should be the case, but what we want to verify is that the user is signed in without starting the login flow? This applies to other tests as well.

Reply:
Can I possibly get some guidance here on how the tests should best be approached? Imo we should check if `auth.service.logout` is only called once while we call `await command.action...` twice per test?

I tried to stub `auth.service.logout` and then check this with `calledonce`, but that doesn't work?

Reply:
Following the spec, we want the command do to nothing if the user is signed in. As such, we should check that `auth.service.logout` hasn't been called. How did you try to stub logout?

Reply:
I've defined a global var and i assign the deactivate function to this global var in the `beforeEach` like this:
`deactivateStub = sinon.stub(auth.connection, 'deactivate').callsFake(() => { });`.

When i run only the test, it says that `deactivateStub` is called once. When I run all the tests, it says that `deactivateStub` is called none 😕